src/api/writer-generator/typescript.ts

```diff
@@ -133,25 +146,103 @@ export class TypeScript extends Writer<TypeScriptOptions> {
         this.lineSM(`import type { ${entities.join(", ")} } from "${tsPackageName}"`);
     }
 
+    private tsProfileClassName(profileName: string): string {
```

Comment:
Just on style, let's move it to function if it doesn’t require class state. Here and the following methods.